pages/content/amp-dev/documentation/guides-and-tutorials/develop/custom-javascript-tutorial.md

```diff
@@ -0,0 +1,262 @@
+
+{{ image('/static/img/docs/tutorials/custom-javascript-tutorial/image1.jpg', 500, 369, layout='intrinsic', alt='Image of basic amp script tutorial starter app', align='' ) }}
+
+After selecting the password input element, the [AMP `on="tap:rules.show"` action]({{g.doc('/content/amp-dev/documentation/guides-and-tutorials/learn/amp-actions-and-events.md', locale=doc.locale).url.path}}?format=websites) is triggered. This reveals the [`<div id="rules" hidden>`]({{g.doc('/content/amp-dev/documentation/guides-and-tutorials/learn/amp-actions-and-events.md', locale=doc.locale).url.path}}?format=websites#*-(all-elements)) element, where our password requirements are listed. 
+
+{{ image('/static/img/docs/tutorials/custom-javascript-tutorial/image2.jpg', 500, 604, layout='intrinsic', alt='Image of basic amp script tutorial starter app with password requirements', align='' ) }}
+
+Play around with different passwords! If you press the submit button before all requirements are met `<amp-form>`’s [`patten`](https://amp.dev/documentation/components/amp-form?format=websites#verification) attribute will throw an error.
+
+{{ image('/static/img/docs/tutorials/custom-javascript-tutorial/image3.jpg', 500, 605, layout='intrinsic', alt='Image of basic amp script tutorial starter app with unhelpful error', align='' ) }}
+
+Frustratingly, it will not explain what is missing. The tool tip only states the required pattern was not met. With the introduction of `<amp-script>` we can build out additional functionality that evaluate the user’s input and communicate what is missing! 
+
+
+# Import `amp-script`
+
+Like nearly all AMP components, `<amp-script>` requires a script tag. Open `index.html` and add the `amp-script` script tag to the head of the document.
+
+
+```html
+<head>
+ ... 
+  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js"></script>
+  ...
+</head>
+
+```
+
+
+The `amp-script` tags must be wrapped around the elements it wishes to mutate and interact with. `amp-script` is unable to mutate or interact with anything that is not a direct child of it.
+
+Our functionality relays and changes the DOM of our `form`. Place the `<form>` element inside the `<amp-script>` component. 
+
+
+```html
+<amp-script src="http://localhost:8080/js/script.js">
+    <form method="post"
+    action-xhr="/form"
+    target="_top"
+    class="card">
+        . . . 
+    </form>
+</amp-script>
+```
+
+[tip type="note"]
+**Note**: Currently, the `src` attribute must point to an absolute URL.
+[/tip]
+
+The `src` attribute points to the filepath `http://localhost:8080/js/script.js`. Create a directory titled `js` and add the `script.js` file. 
```

Comment:
maybe make clear where to create the directory (in your working directory you cloned earlier, or something)